src/macaron/dependency_analyzer/dependency_resolver.py

```diff
@@ -129,6 +130,16 @@ def add_latest_version(
         url_to_artifact: dict[str, set]
             Used to detect artifacts that have similar repos.
         """
+        if item["url"] == "" and item["version"] != "unspecified":
+            gav = f"{item['group']}:{item['name']}:{item['version']}"
+            urls = find_repo(gav, ["scm.url"])
+            if len(urls) != 0:
```

Comment:
```suggestion
            if urls:
```
In Python this if condition would be false if the list is empty.